backend/lib/src/repository/mock.rs

```diff
@@ -0,0 +1,647 @@
+        let id = self.next_id();
+        let now = Utc::now().naive_utc();
+
+        let bucket = Bucket {
+            id,
+            msp_id: new_bucket.msp_id,
+            account: new_bucket.account,
+            onchain_bucket_id: new_bucket.onchain_bucket_id,
+            name: new_bucket.name,
+            collection_id: new_bucket.collection_id,
+            private: new_bucket.private,
+            merkle_root: new_bucket.merkle_root,
+            created_at: now,
+            updated_at: now,
+        };
+
+        buckets.insert(id, bucket.clone());
+        Ok(bucket)
+    }
+
+    // ============ File Write Operations ============
+
+    async fn create_file(&self, new_file: NewFile) -> RepositoryResult<File> {
+        let mut files = self.files.write().await;
+
+        // Check for duplicate key
+        if files.contains_key(&new_file.file_key) {
+            return Err(RepositoryError::Database(
+                diesel::result::Error::DatabaseError(
+                    diesel::result::DatabaseErrorKind::UniqueViolation,
+                    Box::new("File with this key already exists".to_string()),
+                ),
+            ));
+        }
+
+        let id = self.next_id();
+        let now = Utc::now().naive_utc();
+
+        let file = File {
+            id,
+            account: new_file.account,
+            file_key: new_file.file_key.clone(),
+            bucket_id: new_file.bucket_id,
+            location: new_file.location,
+            fingerprint: new_file.fingerprint,
+            size: new_file.size,
+            step: new_file.step,
+            deletion_status: None,
+            created_at: now,
+            updated_at: now,
+        };
+
+        files.insert(file.file_key.clone(), file.clone());
+        Ok(file)
+    }
+
+    async fn update_file_step(&self, file_key: &[u8], step: i32) -> RepositoryResult<()> {
+        let mut files = self.files.write().await;
+        match files.get_mut(file_key) {
+            Some(file) => {
+                file.step = step;
+                file.updated_at = Utc::now().naive_utc();
+                Ok(())
+            }
+            None => Err(RepositoryError::not_found("File")),
+        }
+    }
+
+    async fn delete_file(&self, file_key: &[u8]) -> RepositoryResult<()> {
+        let mut files = self.files.write().await;
+        match files.remove(file_key) {
+            Some(_) => Ok(()),
+            None => Err(RepositoryError::not_found("File")),
+        }
+    }
+
+    async fn clear_all(&self) {
+        let mut bsps = self.bsps.write().await;
+        let mut buckets = self.buckets.write().await;
+        let mut files = self.files.write().await;
+
+        bsps.clear();
+        buckets.clear();
+        files.clear();
+
+        // Reset ID counter
+        self.next_id.store(1, Ordering::SeqCst);
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    //! Scaffolding tests for MockRepository - basic coverage only
+    //!
+    //! Missing coverage:
```

Comment:
```suggestion
    //! TODO: Missing coverage:
```
Just to be able to search the codebase with `TODO:` and find it 😉